Core/automation/lib/javascript/core/log.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Logging functions and variables
+ * 
+ * Copyright (c) 2019 Contributors to the openHAB Scripters project
+ * 
+ * @author Martin Stangl - initial contribution
+ */
+'use strict';	
+
+(function context (context) {
+    'use strict';	
+    
+    load(__DIR__+'/actions.js');        
+
+    context.NOTIFY_OFF = 0;
+    context.NOTIFY_ERROR = 200;
+    context.NOTIFY_WARN = 300;
+    context.NOTIFY_INFO = 400;
+    context.NOTIFY_DEBUG = 500;
+
+    context.Logger = function Logger (name, notificationLevel, config) {        
+        if (name === undefined) {
+            name = Error().stack.split('\n')[2].split('/').slice(-2).join('.').split(':')[0];
+            name = name.slice(-3) == ".js" ? name.slice(0,-3) : null;
+        }
+        var _logger  = Java.type("org.slf4j.LoggerFactory").getLogger(name === null ? "jsr223.javascript" : "jsr223.javascript." + name.toString().toLowerCase());        
+        var _messageFormatter = Java.type("org.slf4j.helpers.MessageFormatter");
+
+        try {
+            // Set default config for config params not provided.
+            if (config === undefined) config = {};
+            if (config.ERROR === undefined) config.ERROR = {};
+            if (config.WARN  === undefined) config.WARN  = {};
+            if (config.INFO  === undefined) config.INFO  = {};
+            if (config.DEBUG === undefined) config.DEBUG = {};
+            if (config.ERROR.prefix === undefined) config.ERROR.prefix = "short";
+            if (config.WARN.prefix  === undefined) config.WARN.prefix  = "none";
+            if (config.INFO.prefix  === undefined) config.INFO.prefix  = "none";
+            if (config.DEBUG.prefix === undefined) config.DEBUG.prefix = "short";       
+
+            return Object.create(Object.prototype, {
+                _notificationLevel: { value: (notificationLevel === undefined || notificationLevel === null) ? context.NOTIFY_OFF : notificationLevel },
```

Comment:
You are using an underscore prefix for private vars. @openhab-5iver is this a coding style that we should be using in general?

Also note that these vars are still accessible; if you want to actually make them inaccessible, I'd suggest that you move the definition of these vars from the object block into the containing function body.

Reply:
Yes, I am aware of that and even did it this way originally.  But then I got worried if this would not cause an issue when I instantiate multiple Loggers as I got confused by all these JS context mechanism. So I just moved it into the returned object, well aware that I loose the privacy.
Maybe I should have just taken the time and test it out. I will do actually do this and move them into the function body, if it works.

Reply:
> @openhab-5iver is this a coding style that we should be using in general?

That would be my preference. That's how we do it in Python, so it fit in perfectly IMO!

Reply:
As long as the vars are _defined_ in the function, they should map 1:1 to the object being returned so you should be fine.

@openhab-5iver I can certainly see the bias for Python here! I agree that striving for alignment is good. I would only caution that there are likely to be areas requiring divergence if you want the JS support to be idiomatic to JS, not Python (although I don't think this is one of them!).

Reply:
Truly, it is not bias. I'm attempting to keep some consistency in place to ease the future migration to a Scripting API, which will absorb pretty much all of the helper library functionality. That will take some time though, and development is continuing. I'm hopeful that, with the new development efforts on the JS libs, there will be some collaboration on how to move forward with bringing their functionality to the level of the Python libs. 

I have done plenty of development, but my background is from the management side of things. I'm currently the only active maintainer in this repo, and I am very open to collaboration on its technical direction, especially with the JS libs!